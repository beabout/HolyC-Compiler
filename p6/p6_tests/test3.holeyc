int b;
int increment(int d){
  d++;
  return d;
}

void fn(int c){
  c = increment(c);
  if(c > 4){
    TOCONSOLE c;
    c = 12;
  }
  char fromc;
  FROMCONSOLE c;
  return; 
}