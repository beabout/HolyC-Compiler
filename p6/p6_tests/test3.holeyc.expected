[BEGIN GLOBALS]
b
[END GLOBALS]
[BEGIN increment LOCALS]
d (formal)
[END increment LOCALS]
lbl_fun_increment: enter increment
getarg 1 [d]
[d] := [d] ADD64 1
setret [d]
goto lbl_0
lbl_0: leave increment
[BEGIN fn LOCALS]
c (formal)
fromc (local)
tmp0 (tmp)
tmp1 (tmp)
[END fn LOCALS]
lbl_fun_fn: enter fn
getarg 1 [c]
setarg 1 [c]
call increment
getres [tmp0]
[c] := [tmp0]
[tmp1] := [c] GT64 4
IFZ [tmp1] GOTO lbl_2
OUTPUT [c]
[c] := 12
lbl_2: nop
INPUT [c]
goto lbl_1
lbl_1: leave fn
